file-formats/doma/type/zif_aff_doma_v1.abap

```diff
@@ -0,0 +1,115 @@
+INTERFACE zif_aff_doma_v1 PUBLIC.
+
+  "! <p class="shorttext">Plus - Minus sign for domains</p>
+  "! Plus - Minus sign for domains
+  TYPES ty_sign                 TYPE c.
+
+  "! <p class="shorttext">Case sensitive flag for domains</p>
+  "! Case sensiive flag for domains
+  TYPES ty_case_sensitive       TYPE abap_bool.
+
+  "! <p class="shorttext">Data type for the domain</p>
+  "! Data type for the domain
+  TYPES ty_data_type            TYPE c LENGTH 4.
+
+  "! <p class="shorttext">No Characters</p>
+  "! Number of Characters
+ "! $minimum 0
+ "! $maximum 999999
+ TYPES ty_numb_char            TYPE i.
+
+ "! <p class="shorttext">Decimals</p>
+ "! Decimals
+ "! $minimum 0
+ "! $maximum 999999
+ TYPES ty_decimals             TYPE i.
+
+ "! <p class="shorttext">Output Length</p>
+ "! Output Length
+ "! $minimum 0
+ "! $maximum 999999
+ TYPES ty_output_length        TYPE i.
+
+  "! <p class="shorttext">Convertion Routine</p>
+  "! Convertion Routine
+  TYPES ty_conv_routine         TYPE c LENGTH 5.
+
+  "! <p class="shorttext">Value table</p>
+  "! Value table
+  TYPES ty_value_table          TYPE c LENGTH 30.
+
+  "! <p class="shorttext">Single Values</p>
+  "! Single Values
+  TYPES:
+    BEGIN OF ty_single_value,
+      "! <p class="shorttext">Fixed value</p>
+      "! Value
+      fixed_value                 TYPE c LENGTH 10,
+      "! <p class="shorttext">Description</p>
+      "! Description
+      description                 TYPE zif_aff_types_v1=>ty_description_60,
+    END OF ty_single_value,
+
+    ty_single_values        TYPE STANDARD TABLE OF ty_single_value WITH DEFAULT KEY.
+
+  "! <p class="shorttext">Interval values</p>
+  "! Interval values
+  TYPES:
+    BEGIN OF ty_intervals_value,
+      "! <p class="shorttext">Low limit of the interval</p>
+      "! Low value for the interval
+      low_limit                   TYPE c LENGTH 10,
+      "! <p class="shorttext">High limit of the interval</p>
+      "! High value for the interval
+      "! $required
+      high_limit                  TYPE c LENGTH 10,
+      "! <p class="shorttext">Description</p>
+      "! Description
+      description                 TYPE zif_aff_types_v1=>ty_description_60,
+    END OF ty_intervals_value,
+
+    ty_intervals_values     TYPE STANDARD TABLE OF ty_intervals_value WITH DEFAULT KEY.
+
+  TYPES:
+    "! <p class="shorttext">Domain Properties</p>
+    "! Domain properties
+    BEGIN OF ty_main,
+      "! $required
+      format_version            TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! header
+      "! $required
+      header                    TYPE zif_aff_types_v1=>ty_header_60,
+
+      "! Plus, minus sign
+      sign                      TYPE ty_sign,
+
+      "! Set if case sensitive
+      case_sensitive            TYPE ty_case_sensitive,
+
+      "! $required
+      data_type                 TYPE ty_data_type,
+
+      "! $required
+      num_chars                 TYPE ty_numb_char,
```

Comment:
I am wondering. Didn't we change this already?

```suggestion
      length                 TYPE ty_length,
```

Reply:
Not that I remember, and I couldn't find anything in the history, unless I missed it :P

Reply:
Maybe, I was wrong. Thanks anyhow :)